resteasy-core/src/main/java/org/jboss/resteasy/util/TypeConverter.java

```diff
@@ -75,14 +75,14 @@ private TypeConverter()
       }
       if (Character.class.equals(targetType))
       {
-         if (source.length() == 0) return targetType.cast(new Character('\0'));
-         return targetType.cast(new Character(source.charAt(0)));
+         if (source.length() == 0) return targetType.cast(Character.valueOf('\0'));
+         return targetType.cast(Character.valueOf(source.charAt(0)));
```

Comment:
The boxing here is not required. It can just be `targetType.case(source.charAt(0))`